compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -1033,6 +1033,16 @@ impl<'a> Parser<'a> {
         if self.eat_keyword(kw::For) {
             self.expect_lt()?;
             let params = self.parse_generic_params()?;
+            let params = params
```

Comment:
This should be done in ast validation or somewhere *after* macro expansion. I'm pretty sure that this PR will cause the following code to go from pass to fail:

```
macro_rules! a {
    ($ty:ty) => {}
}

a! { for<T = &i32> fn() }
```